packages/agent-utils/src/vectors/LocalDocument.ts

```diff
@@ -0,0 +1,90 @@
+import path from "path-browserify";
+import { Workspace } from "../sys";
+
+const VECTOR_FILENAME = "vector.json"
+const METADATA_FILENAME = "metadata.json"
+const DOCUMENT_FILENAME = "document.json"
+
+export type DocumentMetadata = Record<string, any>
+
+export class LocalDocument {
+  constructor(
+    readonly id: string,
+    private config: {
+      uri: string,
+      workspace: Workspace,
+    }
+  ) {}
+
+  vector(): number[] {
+    const vectorPath = path.join(this.config.uri, VECTOR_FILENAME)
+
+    if (!this.config.workspace.existsSync(vectorPath)) {
+      throw new Error(`Vector file for '${this.id}' does not exist: ${vectorPath}. Did you forget to call .save()?`)
+    }
+
+    const vectorFileContent = this.config.workspace.readFileSync(vectorPath)
+    const vector: { id: string; vector: number[]; } = JSON.parse(vectorFileContent)
+
+    return vector.vector
+  }
+
+  text(): string {
+    const documentPath = path.join(this.config.uri, DOCUMENT_FILENAME)
```

Comment:
NIT: there's a naming mismatch between "document" and "text".

Reply:
So, currently document only holds a single `text` property, but the idea was to maybe extend that for other user-defined properties in the future (just like regular DBs). Does that make sense?